### PR TITLE
Easy image selection in configuration

### PR DESCRIPTION
**Pull request by kuhlkrein**
The idea is to make the selection of customized images easier and more visual for users that don't  easily deal with directories and paths, here is how the first idea looks: 

Capture d’écran de 2020-02-28 11-21-32

### CODE REVIEW

**Comment by schwabdidier** [comment]
Instead of maggiccards, Portraits, blocs, I suggest to rename for something like cards, Portraits (?) full-screen or something like that. I created this names in 2016 with the 3 or 4 firsts games, now they use in many other games.

By the way, I love the idea of this new interface !

@Tohaker @YannRobert we will continue to permit upload of a complete zip or folder

**Comment by kuhlkrein** [comment]
I added a thumbnail logo for each section and I think it would be good to probably put the add button before the images preview, and not after

**Comment by kuhlkrein** [comment]
hello @Tohaker ! There is a `java.lang.IllegalStateException: Not on FX application thread;` for each call to  `ConfigurationContext context = new ConfigurationContext(mockGazePlay);` should we specify a thread or do we have to create a better `ConfigurationContext mockContext` ? I didn not change anything like this in the tests so I don't know why I have these errors...

**Comment by Tohaker** [comment]
Whenever that happens, you need to run the test on the FX thread with the following:
```
Platform.runLater(() -> {
     // your code here
}
```
You'll also need to use a handy test util  I made called `TestingUtils.waitForRunLater();` after the `runLater()` that will ensure the test doesn't finish early. If the test can't find the TestingUtils class, you will need to import it in the `build.gradle` file for that module. You can find an example in the `gazeplay-commons` build file.

**Comment by kuhlkrein** [comment]
Test are working fine localy but there is the same problem here, looks like it is using old version of test file...

**Comment by kuhlkrein** [comment]
Some people had the same problem but I don't get it... do you understand @YannRobert or @Tohaker  ? 
link to other people git issue having same proble:
https://github.com/actions/checkout/issues/23

**Comment by kuhlkrein** [comment]
Ah so it's actually the action repo....

**Comment by Tohaker** [comment]
Not a lot we can do if they've broken an action. We'll just have to wait until they push a fix for it.

**Comment by kuhlkrein** [comment]
@Tohaker now that I moved from draft to normal PR it looks good but still, there is a problem with the tests in the build

**Comment by kuhlkrein** [comment]
@Tohaker , @Tohaker they said in the issue mentioned above that it should be fixed by switching to actions/checkout@v2, is it possible here? I don't know if it changes something for us

**Comment by Tohaker** [comment]
Just change it to v2 in the ci.yml anc cd.yml and see what happens. It's just a git checkout, it shouldn't break anything

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/configuration/CustomFileChooser.java [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+    }
+
+    private I18NButton createAddButton(int flowPaneIndex) {
+        I18NButton add = new I18NButton(translator, "addNewImages");
+        add.setPrefHeight(10);
+        add.setOnAction(e -> {
+            final JFileChooser fileChooser = new JFileChooser();
+            fileChooser.setMultiSelectionEnabled(true);
+            fileChooser.showOpenDialog(new JFrame());
+            File[] files = fileChooser.getSelectedFiles();
```
[the remark this answers is not in the document]
Comment (kuhlkrein) [reply gazeplay/src/main/java/net/gazeplay/ui/scenes/configuration/CustomFileChooser.java]: The problem with JavaFX file chooser is that it allows the selection of only one file at a time... :/

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/configuration/CustomFileChooser.java [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+        final Button no = new Button(translator.translate("NoCancel"));
+        no.getStyleClass().add("gameChooserButton");
+        no.getStyleClass().add("gameVariation");
+        no.getStyleClass().add("button");
+        no.setMinHeight(gazePlay.getPrimaryStage().getHeight() / 10);
+        no.setMinWidth(gazePlay.getPrimaryStage().getWidth() / 10);
+        no.addEventFilter(MouseEvent.MOUSE_CLICKED, (EventHandler<Event>) event -> {
+            this.getScene().getRoot().setEffect(null);
+            dialog.close();
+        });
```
[the remark this answers is not in the document]
Comment (kuhlkrein) [reply gazeplay/src/main/java/net/gazeplay/ui/scenes/configuration/CustomFileChooser.java]: The part 
```
this.getScene().getRoot().setEffect(null);
dialog.close();
```
right ?

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/configuration/CustomFileChooser.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kuhlkrein) [reply gazeplay/src/main/java/net/gazeplay/ui/scenes/configuration/CustomFileChooser.java]: Ah ok I managed to do this, my search on it was probably bad last time

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/configuration/CustomFileChooser.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tohaker) [reply gazeplay/src/main/java/net/gazeplay/ui/scenes/configuration/CustomFileChooser.java]: No sorry I mean the whole button creation, involving adding styles and stuff

[the remark this answers is not in the document]
Comment (kuhlkrein) [reply gazeplay/src/main/java/net/gazeplay/ui/scenes/configuration/CustomFileChooser.java]: Ah Yeah ok sure

**Comment by schwabdidier** [comment]
@kuhlkrein please resolve new conflicts

**Comment by kuhlkrein** [comment]
@schwabdidier done, sorry it was nothing important at all, just the date of the gazeplay property that changed

**Pull request merged**